Makefile

```diff
@@ -133,8 +133,9 @@ test-e2e: $(TEST_E2E_DEPS)
 test-unit: use-main
 	@go test -v -race $(shell go list ./... | grep -v tests/)
 
+## TODO(nikhil): address data-race in integration-test runs
```

Comment:
let's make issue for this?

It could just be an issue with the parallel testing